geonode/base/api/serializers.py

```diff
@@ -278,15 +280,51 @@ class DownloadLinkField(DynamicComputedField):
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
+    @deprecated(version="4.2.0", reason="Will be replaced by download_urls")
     def get_attribute(self, instance):
         try:
+            logger.info(
+                "This field is deprecated, and will be removed in the future GeoNode version. Please refer to download_urls"
+            )
             _instance = instance.get_real_instance()
             return _instance.download_url if hasattr(_instance, "download_url") else None
         except Exception as e:
             logger.exception(e)
             return None
 
 
+class DownloadArrayLinkField(DynamicComputedField):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def get_attribute(self, instance):
+        try:
+            _instance = instance.get_real_instance()
+        except Exception as e:
+            logger.exception(e)
+            raise e
+        download_urls = []
+        if _instance.resource_type in ["map"] + get_geoapp_subtypes():
+            return []
+        elif _instance.resource_type in ["document"]:
+            return [
+                {
+                    "url": _instance.download_url,
+                    "ajax_safe": _instance.download_is_ajax_safe,
+                }
+            ]
+        elif _instance.resource_type in ["dataset"]:
+            for item in settings.DATASET_DOWNLOAD_HANDLERS:
+                handler = import_string(item)
```

Comment:
IMHO the import should be done at import time, not inside the method call where it will be called each time the field is requested.